### PR TITLE
Add name() and name_with_middle() fn

### PR DESCRIPTION
**Pull request by sibeizhou**
And modified some use of the function related

### CODE REVIEW

File: src/locales/en/food/dish_pattern.rs [hunk 1]
```diff
@@ -7,17 +7,7 @@ use super::meat::meat;
...
-
-	match format {
-        0 => male_first_name(),
-        _ => female_first_name(),
-    }
-}
+use crate::locales::en::person::name::first_name;
 
 #[wasm_bindgen(js_name = food_dish_parttern)]
 pub fn dish_parttern() -> String {
```
Comment (clearfeld) [inline src/locales/en/food/dish_pattern.rs:13]: Typo `parttern` instead of `pattern`

File: src/locales/en/location/city_parttern.rs [hunk 2]
```diff
@@ -8,17 +8,14 @@ use crate::locales::en::person::name;
 
 #[wasm_bindgen(js_name = location_city_parttern)]
 pub fn city_parttern() -> String {
```
Comment (clearfeld) [inline src/locales/en/location/city_parttern.rs]: Typo and in the name of the file

File: src/locales/en/location/street_parttern.rs [hunk 3]
```diff
@@ -6,14 +6,13 @@ use crate::locales::en::person::name;
 
 #[wasm_bindgen(js_name = location_street_parttern)]
 pub fn street_parttern() -> String {
```
Comment (clearfeld) [inline src/locales/en/location/street_parttern.rs]: Same here

**Review by clearfeld — CHANGES_REQUESTED** [review]

**Comment by sibeizhou** [comment]
Already fixed the typo mistakes

**Review by clearfeld — APPROVED** [review]

**Pull request merged**